test/unit/configuration/configuration_parser_test.js

```diff
@@ -0,0 +1,139 @@
+const assert = require('assert');
+const configurationParser = require('./../../../lib/configuration/configuration_parser');
+const fsPromises = require('fs/promises');
+const os = require('os');
+const path = require('path');
+const { v4: uuidv4 } = require('uuid');
+const tempDir = path.join(os.tmpdir(), uuidv4());
+
+describe('Configuration parsing tests', function () {
+
+  before(async function () {
+    await fsPromises.mkdir(tempDir, { recursive: false });
+  });
+
+  after(async function () {
+    await fsPromises.rm(tempDir, { recursive: true, force: true });
+  });
+
+  it('should parse json', async function () {
+    // given
+    const fileName = 'config.json';
+    const filePath = path.join(tempDir, fileName);
+    const fileContent = `{
+            "common": {
+                "log_level": "info",
+                "log_path": "/some-path/some-directory"
+            } 
+        }`;
+    await fsPromises.writeFile(filePath, fileContent, { encoding: 'utf8' });
+
+    // when
+    const configuration = await configurationParser.parse(filePath);
+
+    // then
+    assert.equal(configuration.logLevel, 'info');
+    assert.equal(configuration.logPath, '/some-path/some-directory');
+  });
+
+  it('should fail when wrong level', async function () {
+    // given
+    const fileName = 'config_wrong_level.json';
+    const filePath = path.join(tempDir, fileName);
+    const fileContent = `{
+            "common": {
+                "log_level": "unknown",
+                "log_path": "/some-path/some-directory"
+            } 
+        }`;
+    await fsPromises.writeFile(filePath, fileContent, { encoding: 'utf8' });
+
+    // expect
+    await assert.rejects(
+      async () => await configurationParser.parse(filePath),
+      (err) => {
+        assert.strictEqual(err.name, 'Error');
+        assert.strictEqual(err.message, 'Parsing easy logging configuration failed');
```

Comment:
we are loosing information why parsing failed - user won't know is it because of permissions or missing path or wrong name of log level

Reply:
changed